test/integration/http2_integration_test.cc

```diff
@@ -384,6 +384,63 @@ TEST_P(Http2IntegrationTest, SimultaneousRequestWithBufferLimits) {
   simultaneousRequest(1024 * 32, 1024 * 16);
 }
 
+// Test downstream connection delayed close processing.
+TEST_P(Http2IntegrationTest, DelayedCloseAfterBadFrame) {
+  initialize();
+  Buffer::OwnedImpl buffer("PRI * HTTP/2.0\r\n\r\nSM\r\n\r\nhelloworldcauseanerror");
+  std::string response;
+  RawConnectionDriver connection(
+      lookupPort("http"), buffer,
+      [&](Network::ClientConnection& connection, const Buffer::Instance& data) -> void {
+        response.append(data.toString());
+        connection.dispatcher().exit();
+      },
+      version_);
+
+  connection.run();
+  EXPECT_TRUE(response.find("SETTINGS expected") != std::string::npos);
```

Comment:
I find the form `EXPECT_THAT(response, HasSubstr("SETTINGS expected"))` more readable when trying to understand how the test works.

Reply:
Thanks for pointing out the matcher. Done.